### PR TITLE
Refactor where field eval by using fromIdentVisitor.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**

### CODE REVIEW

File: plan/plans/where_test.go [hunk 1]
```diff
@@ -44,7 +44,9 @@ func (t *testWhereSuit) TestWhere(c *C) {
 		Expr: &expression.BinaryOperation{
 			Op: opcode.GE,
 			L: &expression.Ident{
-				CIStr: model.NewCIStr("id"),
+				CIStr:      model.NewCIStr("id"),
+				ReferScope: 2,
```
Comment (ngaut) [inline plan/plans/where_test.go]: Why use magic number 2 ?

File: plan/plans/where_test.go [hunk 2]
```diff
@@ -44,7 +44,9 @@ func (t *testWhereSuit) TestWhere(c *C) {
...
+				ReferIndex: 0,
```
Comment (ngaut) [inline plan/plans/where_test.go]: Why use magic number 0 ?

File: plan/plans/where_test.go (hunk 1 above)
Comment (qiuyesuifeng) [inline plan/plans/where_test.go]: Gotcha, i will use a const value.

File: plan/plans/where_test.go (hunk 2 above)
Comment (qiuyesuifeng) [inline plan/plans/where_test.go]: 0 means the index in table from fields, here is `[id, name]`, so the ReferIndex value is 0.

**Pull request merged**